### PR TITLE
Add TF generate sample tests with all logit processors

### PR DESCRIPTION
**Pull request by gante**
# What does this PR do?

This PR adds a GPT2 TF generate sample test with all logit processors. It is a requirement for the TF generate sample refactor (https://github.com/huggingface/transformers/pull/15793), which is part of the TF generate refactor.

### CODE REVIEW

File: tests/gpt2/test_modeling_tf_gpt2.py [hunk 1]
```diff
@@ -468,6 +468,37 @@ def test_lm_generate_distilgpt2_batch_special(self):
...
+        input_ids = tokenizer(sentences, return_tensors="tf", padding=True).input_ids
+
+        generation_kwargs = {
+            "do_sample": True,
+            "bad_words_ids": [tokenizer("is").input_ids, tokenizer("angry about").input_ids],
+            "no_repeat_ngram_size": 2,
+            "repetition_penalty": 1.3,
+            "temperature": 1.5,
+            "top_k": 500,
+            "top_p": 0.9,
```
Comment (gante) [inline tests/gpt2/test_modeling_tf_gpt2.py:489]: Changing one of these 3 will change the output, as expected.

**Review by Rocketknight1 — APPROVED** [review]
This looks good! Is the plan to add a single specific test for each logit processor, as they're converted?

**Comment by Rocketknight1** [comment]
Also, code quality is failing, be sure to `make style`!

**Comment by gante** [comment]
> Is the plan to add a single specific test for each logit processor, as they're converted?

Yeah, in the other PR (master doesn't have them yet)

**Comment by patrickvonplaten** [comment]
Perfect! Sorry to be a bit annoying here - could you also add one random test for T5? I know that this is not a very common use case for encoder-decoder settings, but the encoder-decoder setting is significantly different from decoder-only so that it would be good to be sure that nothing is broken there before the refactor. Think you can take more or less the exact same input that you used for GPT2 (the output might not be that sensible, but anyways good to test encoder-decoder in sample mode).

**Comment by gante** [comment]
@patrickvonplaten now with a T5 generate sample test. I took the liberty to find a combination of inputs where the output of one is very unstable (the first one), and the output of the other is pretty stable (the second one). It may help us distinguish the case where the new generate is completely different from the case where there are minor numerical differences.

**Review by patrickvonplaten — APPROVED** [review]

**Pull request merged**